projects/hello_adc/src/hello_adc.c

```diff
@@ -0,0 +1,39 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+#define BUTTON_PIN 178
+#define READ_FROM_PIN 166
+
+GpioItCallback prv_log_pin_data(GpioAddress *button_addr, void *context);
+
+int main() {
+  interrupt_init();
+  soft_timer_init();
+  gpio_init();
+  GpioAddress button_addr = { GPIO_PORT_A, BUTTON_PIN };
+  GpioAddress read_addr = { GPIO_PORT_A, READ_FROM_PIN };
+  // Setting pin value is not necessary
+  GpioSettings null_settings = {};
+  gpio_init_pin(&button_addr, &null_settings);
+  gpio_init_pin(&read_addr, &null_settings);
+  adc_init(ADC_MODE_SINGLE);
+  InterruptSettings button_interrupt = { INTERRUPT_TYPE_EVENT, INTERRUPT_PRIORITY_NORMAL };
```

Comment:
FYI: it's alright for small one liners, but we generally prefer the GNU struct initialization syntax like:
```c
InterruptSettings button_interrupt = {
  .type = INTERRUPT_TYPE_EVENT,
  .priority = INTERRUPT_PRIORITY_NORMAL,
};
```